kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/CrdValidationIT.java

```diff
@@ -0,0 +1,135 @@
+import io.fabric8.kubernetes.api.model.Namespace;
+import io.fabric8.kubernetes.api.model.NamespaceBuilder;
+import io.fabric8.kubernetes.api.model.Status;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientException;
+import io.fabric8.kubernetes.client.dsl.NamespaceableResource;
+import io.fabric8.kubernetes.client.dsl.NonDeletingOperation;
+import io.javaoperatorsdk.operator.junit.LocallyRunOperatorExtension;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxyIngress;
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaService;
+import io.kroxylicious.kubernetes.api.v1alpha1.VirtualKafkaCluster;
+import io.kroxylicious.kubernetes.filter.api.v1alpha1.KafkaProtocolFilter;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@EnabledIf(value = "io.kroxylicious.kubernetes.operator.OperatorTestUtils#isKubeClientAvailable", disabledReason = "no viable kube client available")
+class CrdValidationIT {
+
+    public static final Namespace NAMESPACE = new NamespaceBuilder().withNewMetadata().withName("proxy-ns").endMetadata().build();
+    public static final ObjectMapper YAML_MAPPER = new ObjectMapper(new YAMLFactory());
+
+    @BeforeAll
+    static void beforeAll() {
+        KubernetesClient client = OperatorTestUtils.kubeClient();
+        LocallyRunOperatorExtension.applyCrd(KafkaProtocolFilter.class, client);
+        LocallyRunOperatorExtension.applyCrd(KafkaProxy.class, client);
+        LocallyRunOperatorExtension.applyCrd(VirtualKafkaCluster.class, client);
+        LocallyRunOperatorExtension.applyCrd(KafkaService.class, client);
+        LocallyRunOperatorExtension.applyCrd(KafkaProxyIngress.class, client);
+        client.namespaces().resource(NAMESPACE).createOr(NonDeletingOperation::update);
+    }
+
+    @AfterAll
+    static void afterAll() {
+        try (KubernetesClient kubernetesClient = OperatorTestUtils.kubeClient()) {
+            kubernetesClient.namespaces().resource(NAMESPACE).delete();
+            kubernetesClient.resources(KafkaProtocolFilter.class).delete();
+            kubernetesClient.resources(KafkaProxyIngress.class).delete();
+            kubernetesClient.resources(KafkaProxy.class).delete();
+            kubernetesClient.resources(VirtualKafkaCluster.class).delete();
+            kubernetesClient.resources(KafkaService.class).delete();
+        }
+    }
+
+    public static Stream<Path> testDerivedResourceInputsValid() {
+        return TestFiles.recursiveFilesInDirectoryForTest(DerivedResourcesTest.class, "in-*.yaml").stream();
+    }
+
+    public static Stream<Arguments> testResourceInvalid() {
+        return TestFiles.recursiveFilesInDirectoryForTest(CrdValidationIT.class, "invalid-*.yaml").stream().map(p -> {
+            try {
+                return Arguments.of(p.toString(), YAML_MAPPER.readValue(p.toFile(), InvalidResource.class));
```

Comment:
Using `Arguments.argumentSet()` will allow you to drop the `(name = "{0}")` from `@ParameterizedTest` and also the ignored name parameter.